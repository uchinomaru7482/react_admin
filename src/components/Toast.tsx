import React, { useContext } from 'react'
import { BiBell, BiX } from 'react-icons/bi'

import { ToastContext, ToastKind } from '../contexts/ToastContext'

type Props = {
  toastType: string
  indexNum: number
}

const Toast: React.VFC<Props> = (props) => {
  let content = ''
  let border = ''
  let bg = ''
  let text = ''
  if (props.toastType === ToastKind.Success) {
    content = 'Execution completed.'
    border = 'border-indigo-300'
    bg = 'bg-indigo-100'
    text = 'text-indigo-700'
  } else if (props.toastType === ToastKind.Error) {
    content = 'Error has occurred.'
    border = 'border-pink-300'
    bg = 'bg-pink-100'
    text = 'text-pink-700'
  }
  const toast = useContext(ToastContext)
  const removeToast = () => toast.removeToast(props.indexNum)

  return (
    <>
      <div className='toast'>
        <div
          className={`${border} shadow-lg w-96 max-w-full text-sm rounded-lg block border`}
        >
          <div
            className={`${border} ${bg} flex justify-between items-center py-2 px-3 border-b rounded-t-lg`}
          >
            <p className={`${text} font-bold flex items-center`}>
              <BiBell className='h-5 w-5 mr-2' />
              {props.toastType}
            </p>
            <button
              type='button'
              onClick={removeToast}
              className={`${text} w-5 h-5 ml-2 opacity-50 focus:opacity-100 hover:opacity-75`}
            >
              <BiX className='h-5 w-5' />
            </button>
          </div>
          <div className={`${bg} ${text} px-3 py-4 rounded-b-lg break-words`}>
            {content}
          </div>
        </div>
      </div>
    </>
  )
}

export default Toast
