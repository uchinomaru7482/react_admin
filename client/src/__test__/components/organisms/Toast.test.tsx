import { render } from '@testing-library/react'

import Toast from '../../../components/organisms/Toast'

describe('Rendering', () => {
  test('Toast snap shot test', () => {
    const { asFragment } = render(<Toast toastType='Success' indexNum={1} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('Toast type Success', () => {
    const { container } = render(<Toast toastType='Success' indexNum={1} />)
    expect(container.innerHTML).toMatch('Execution completed.')
    expect(
      container.getElementsByClassName('border-indigo-300').length
    ).toEqual(2)
  })

  test('Toast type Error', () => {
    const { container } = render(<Toast toastType='Error' indexNum={1} />)
    expect(container.innerHTML).toMatch('Error has occurred.')
    expect(container.getElementsByClassName('border-pink-300').length).toEqual(
      2
    )
  })
})
