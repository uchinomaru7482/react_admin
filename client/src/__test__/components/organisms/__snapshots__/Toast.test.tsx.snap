// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Toast snap shot test 1`] = `
<DocumentFragment>
  <div
    class="toast"
  >
    <div
      class="border-indigo-300 shadow-lg w-96 max-w-full text-sm rounded-lg block border"
    >
      <div
        class="border-indigo-300 bg-indigo-100 flex justify-between items-center py-2 px-3 border-b rounded-t-lg"
      >
        <p
          class="text-indigo-700 font-bold flex items-center"
        >
          <svg
            class="h-5 w-5 mr-2"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 13.586V10c0-3.217-2.185-5.927-5.145-6.742C13.562 2.52 12.846 2 12 2s-1.562.52-1.855 1.258C7.185 4.074 5 6.783 5 10v3.586l-1.707 1.707A.996.996 0 0 0 3 16v2a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1v-2a.996.996 0 0 0-.293-.707L19 13.586zM19 17H5v-.586l1.707-1.707A.996.996 0 0 0 7 14v-4c0-2.757 2.243-5 5-5s5 2.243 5 5v4c0 .266.105.52.293.707L19 16.414V17zm-7 5a2.98 2.98 0 0 0 2.818-2H9.182A2.98 2.98 0 0 0 12 22z"
            />
          </svg>
          Success
        </p>
        <button
          class="text-indigo-700 w-5 h-5 ml-2 opacity-50 focus:opacity-100 hover:opacity-75"
          type="button"
        >
          <svg
            class="h-5 w-5"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m16.192 6.344-4.243 4.242-4.242-4.242-1.414 1.414L10.535 12l-4.242 4.242 1.414 1.414 4.242-4.242 4.243 4.242 1.414-1.414L13.364 12l4.242-4.242z"
            />
          </svg>
        </button>
      </div>
      <div
        class="bg-indigo-100 text-indigo-700 px-3 py-4 rounded-b-lg break-words"
      >
        Execution completed.
      </div>
    </div>
  </div>
</DocumentFragment>
`;
